go/mysql/flavor_mysqlgr.go

```diff
@@ -238,9 +238,12 @@ func (mysqlGRFlavor) baseShowTablesWithSizes() string {
 	return TablesWithSize80
 }
 
-// supportsFastDropTable is part of the Flavor interface.
-func (mysqlGRFlavor) supportsFastDropTable(c *Conn) (bool, error) {
-	return false, nil
+// supportsCapability is part of the Flavor interface.
+func (mysqlGRFlavor) supportsCapability(serverVersion string, capability FlavorCapability) (bool, error) {
```

Comment:
technically GR is orthogonal to the MySQL version. However, the fact that GR is a flavor will mean that none of the new features can be used with GR.
@y5w why is this flavor required? where is it useful?

Reply:
> the fact that GR is a flavor will mean that none of the new features can be used with GR.

Could you elaborate more what does this mean and what kind of "feature" are you having in mind?

It is a flavor so that we can override status (for vtgate's health check) based on different mysql query

Reply:
> Could you elaborate more what does this mean and what kind of "feature" are you having in mind?

@y5w per "flavor" we might want to determine:

- does the database support INSTANT DDL?
- does it support transactional `gtid_executed`?
- does it support JSON columns?
- does it support the `BACKUP` command?
- etc.

I'm unaware where `mysqlGRFlavor` is used, but please see this followup PR which implements capabilities for `mysqlGRFlavor`: https://github.com/vitessio/vitess/pull/10451

Reply:
@y5w thanks for explaining. I'l look into the refactor later on, thankyou